power_grid_model_c/power_grid_model/include/power_grid_model/main_model.hpp

```diff
@@ -241,8 +243,14 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
             // get component
             // either using ID via hash map
             // either directly using sequence id
-            CompType& comp = has_sequence_id ? components_.template get_item<CompType>(sequence_idx[seq])
-                                             : components_.template get_item<CompType>(it->id);
+            Idx2D const sequence_single =
+                has_sequence_id ? sequence_idx[seq] : components_.template get_idx_by_id<CompType>(it->id);
+
+            if constexpr (CacheType::cached_update) {
```

Comment:
If you use `std::true_type` and `std::false_type`.

```suggestion
            if constexpr (CacheType::value) {
```